### PR TITLE
Expand/clarify synchronous query usage docs.

### LINKED ISSUE #1551: Cannot reload BigQuery job status or results in sync queries
- The documentation [here](https://googlecloudplatform.github.io/gcloud-python/latest/bigquery-usage.html#querying-data-synchronous) is not valid as it uses an undeclared variable (`job`) and a function that doesn't exist in sync queries (`query.reload()`).
 - When I tried to use `query.job` I found each time it returns a different job instance, which is confusing.
 - After caching `job = query.job` in a variable I was able to use `job.state != 'DONE'` and `job.reload()` to keep track of the job status instead of `query.completed` like documented (literally `job.completed`); but I had to call `query.fetch_data()` to manually extract the rows from the query. This makes the `Query` class a simple factory method to build new jobs.

I'm not sure if this is only a documentation issue or if the `QueryJob` class lacks some important methods to be more than a factory for new jobs.

**Original code in the documentation:**
```python
    query = """
      SELECT count(*) AS age_count FROM dataset_name.person_ages
    """
    query = client.run_sync_query(query)
    query.timeout_ms = 1000
    query.run()  # API request
    retry_count = 100
    while retry_count > 0 and not job.complete:
      retry_count -= 1
      time.sleep(10)
      query.reload()  # API request
    print query.rows
```

**Modified code:**
```python
    q = """
      SELECT count(*) AS age_count FROM dataset_name.person_ages
    """
    query = client.run_sync_query(q)
    query.timeout_ms = 1000
    query.run()  # API request

    job = query.job

    retry_count = 100
    while retry_count > 0 and not job.state != 'DONE':
      retry_count -= 1
      time.sleep(10)
      job.reload()  # API request

    (rows, _, _) = query.fetch_data()
    print rows
```

### PR DESCRIPTION
**Pull request by tseaver**
Address three separate cases:

- Query finishes within timeout, with all rows present.

- Query finishes within timeout, but result set is too large for initial response.

- Query times out.

Closes #1551.

### CODE REVIEW

File: docs/bigquery-usage.rst [hunk 1]
```diff
@@ -291,26 +291,88 @@ Run a query which can be expected to complete within bounded time:
...
    SELECT count(*) AS age_count FROM dataset_name.person_ages
    """
-   >>> query = client.run_sync_query(query)
+   >>> query = client.run_sync_query(QUERY)
    >>> query.timeout_ms = 1000
    >>> query.run()  # API request
+   >>> query.complete
+   True
+   >>> len(query.schema)
+   0
```
Comment (dhermes) [inline docs/bigquery-usage.rst]: `0` means that `query.schema[0]` would `IndexError`

File: docs/bigquery-usage.rst [hunk 2]
```diff
@@ -291,26 +291,88 @@ Run a query which can be expected to complete within bounded time:
...
+   """
+   >>> query = client.run_sync_query(QUERY)
+   >>> query.timeout_ms = 1000
+   >>> query.run()  # API request
+   >>> query.complete
+   True
+   >>> query.total_rows
+   1234
+   >>> query.page_token
+   'TOKEN'
```
Comment (dhermes) [inline docs/bigquery-usage.rst]: Make it clear this is a bogus base64 value?

File: docs/bigquery-usage.rst [hunk 3]
```diff
@@ -291,26 +291,88 @@ Run a query which can be expected to complete within bounded time:
...
+If the rows returned by the query do not fit into the inital response,
+then we need to fetch the remaining rows via ``fetch_data``:
+
+.. doctest::
+
+   >>> from gcloud import bigquery
+   >>> client = bigquery.Client()
+   >>> QUERY = """\
+   SELECT * FROM dataset_name.person_ages
+   """
```
Comment (dhermes) [inline docs/bigquery-usage.rst]: Not trying to be a stickler, but just typed this in `python` interpreter:

```python
>>> QUERY = """\
... SELECT * FROM dataset_name.person_ages
... """
```

File: docs/bigquery-usage.rst [hunk 4]
```diff
@@ -291,26 +291,88 @@ Run a query which can be expected to complete within bounded time:
...
+   >>> query.timeout_ms = 1000
+   >>> query.run()  # API request
+   >>> query.complete
+   True
+   >>> query.total_rows
+   1234
+   >>> query.page_token
+   'TOKEN'
+   >>> do_something_with(query.schema, query.rows)
+   >>> token = query_page_token  # for initial request
```
Comment (dhermes) [inline docs/bigquery-usage.rst]: `query.page_token`?

File: docs/bigquery-usage.rst [hunk 5]
```diff
@@ -291,26 +291,88 @@ Run a query which can be expected to complete within bounded time:
...
+   >>> while true:
```
Comment (dhermes) [inline docs/bigquery-usage.rst]: 1. `while True`
1. In general, prefer not to encourage `while True`, instead give some kind of max iterations (or at least make a remark)

Comment (tseaver) [inline docs/bigquery-usage.rst]: I disagree:  this is a "do ... while" loop, for which `while True:`  with a test-and-break is the idiomatic Python spelling (see the rejected [PEP 315](https://www.python.org/dev/peps/pep-0315/)).  It is a different case than a "retry/backoff", where a counter would be needed.

Comment (dhermes) [inline docs/bigquery-usage.rst]: Yeah it's fine, just capitialize your `true`

Comment (tseaver) [inline docs/bigquery-usage.rst]: D'oh!

**Comment by dhermes** [comment]
LGTM.

Grumble grumble something something #212

**Pull request merged**